controllers/windowsmachine_controller.go

```diff
@@ -448,31 +452,31 @@ func (r *WindowsMachineReconciler) validateUserData(privateKey []byte) error {
 
 // isAllowedDeletion determines if the number of machines after deletion of the given machine doesn`t fall below the
 // minHealthyCount
-func (r *WindowsMachineReconciler) isAllowedDeletion(machine *mapi.Machine) bool {
+func (r *WindowsMachineReconciler) isAllowedDeletion(machine *mapi.Machine) (bool, error) {
 	if len(machine.OwnerReferences) == 0 {
-		return false
+		return false, fmt.Errorf("machine has no owner reference")
```

Comment:
can use errors.New here?

Reply:
+1 to `errors.New` and please use `Machines`